docs/rules/jsx-require-react.md

```diff
@@ -0,0 +1,73 @@
+# Require React import in .jsx and .tsx files. (jsx-require-react)
+
+JSX really just provides syntactic sugar for React's `createElement(component, props, ...children)` function. For example:
+
+```jsx
+<Button color="blue">
+  Click Me
+</Button>
+```
+
+will compile to:
+
+```js
+React.createElement(
+  Button,
+  {color: 'blue'},
+  'Click Me'
+)
+```
+
+Built-in components and self-closing tags compile in the same way. For example:
+
+```jsx
+<div className="fancy" />
+```
+
+will compile to:
+
+```js
+React.createElement(
+  'div',
+  {className: 'fancy'},
+  null
+)
+```
+
+## Rule Details
+
+The requirement to import React can easily be missed because it is not directly referenced when authoring JSX. This rule necessitates that React is imported when working in a .tsx or .jsx files.
```

Comment:
Can it be easily missed? I thought you always got an error from somewhere about this already :S

Reply:
Agreed, the build or the editor itself would probably tell you there's an issue before even running into a warning from ESLint.